app/controllers/incomeSources/add/BusinessReportingMethodController.scala

```diff
@@ -87,54 +91,74 @@ class BusinessReportingMethodController @Inject()(val authenticate: Authenticati
 
   }
 
-  private def handleRequest(isAgent: Boolean, id: String)
+  private def handleRequest(isAgent: Boolean)
                            (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = {
 
-    val postAction: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessReportingMethodController.submitAgent(id) else
-      controllers.incomeSources.add.routes.BusinessReportingMethodController.submit(id)
-    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
-    val redirectUrl: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessAddedObligationsController.showAgent(id) else
-      controllers.incomeSources.add.routes.BusinessAddedObligationsController.show(id)
+    sessionService.get(SessionKeys.incomeSourceId).flatMap {
```

Comment:
please avoid nested pattern matching  
Use code provided below, as next code is not really flexible and would be hard to support